src/com/sap/piper/TemporaryCredentialsUtils.groovy

```diff
@@ -0,0 +1,80 @@
+package com.sap.piper
+
+class TemporaryCredentialsUtils implements Serializable {
+
+    private static Script script
+
+    TemporaryCredentialsUtils(Script script) {
+        this.script = script
+    }
+
+    void handleTemporaryCredentials(List credentialItems, String credentialsDirectory, Closure body) {
+        final String credentialsFileName = 'credentials.json'
+
+        if (credentialsDirectory == null) {
+            script.error("This should not happen: Directory for credentials file not specified.")
+        }
+
+        try {
+            if (credentialItems != null && !credentialItems.isEmpty()) {
+                writeCredentials(credentialItems, credentialsDirectory, credentialsFileName)
+            }
+            body()
+        }
+        finally {
+            if (credentialItems != null && !credentialItems.isEmpty()) {
+                deleteCredentials(credentialsDirectory, credentialsFileName)
+            }
+        }
+    }
+
+    private void writeCredentials(List credentialItems, String credentialsDirectory, String credentialsFileName) {
+        if (credentialItems == null || credentialItems.isEmpty()) {
+            script.echo "Not writing any credentials."
+            return
+        }
+
+        assertSystemsFileExists(credentialsDirectory)
+
+        String credentialJson = readCredentials(credentialItems)
+
+        script.echo "Writing credential file with ${credentialItems.size()} items."
+        script.dir(credentialsDirectory) {
+            script.writeFile file: credentialsFileName, text: credentialJson
+        }
+    }
+
+    private void deleteCredentials(String credentialsDirectory, String credentialsFileName) {
+        script.echo "Deleting credential file."
+        script.dir(credentialsDirectory) {
+            script.sh "rm -f ${credentialsFileName}"
+        }
+    }
+
+    private String readCredentials(List credentialItems) {
```

Comment:
```suggestion
    private String encodeToJSON(List credentialItems) {
```
?

Reply:
Maybe `returnCredentialsAsJSON` is reflecting better what the function does? Since, it is actually doing two things (retrieving the credentials with the given ids from jenkins and then return it as JSON string).

Reply:
Sure.